docs/source/developers/continuous_integration/overview.rst

```diff
@@ -0,0 +1,66 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+Continuous Integration
+======================
+
+Continuous Integration for Arrow is fairly complex as it needs to run across different combinations of package managers, compilers, versions of multiple sofware libraries,  operating systems, and other potential sources of variation.  In this article, we will give an overview of its main components and the relevant files and directories.
+
+Two files central to Arrow CI are:
+
+* `docker-compose.yml` - here we define docker services which can be configured using either enviroment variables, or the default values for these variables.
+* `.env` - here we define default values to configure the services in `docker-compose.yml`
+
+One thing to note is the some of the services defined in `docker-compose.yml` are interdependent.  When running a services locally, you must either manually build its dependencies first, or build it via the use of `archery run ...` which automatically finds and builds dependencies.  For more information on archery, see <link>.
+
+There are numerous important directories in the Arrow project which relate to CI:
+
+* `.github/worflows` - workflows that are run via GitHub actions and are triggered by things like pull requests being submitted or merged
+* `dev/tasks` - containing jobs which are run via Archery and Crossbow, typically nightly builds or relating to the release process
+* `ci/` - containing scripts supporting the various builds
+
+Instead of thinking about Arrow CI in terms of files and folders, it may be conceptually simpler to instead divide it into 2 main categories:
+
+* CI jobs which are triggered based on specific actions on GitHub (pull requests opened, pull requests merged, etc)
+* Builds which are manually triggered on a nightly basis or via Archery
+
+GitHub Actions builds
+-----------------------
+
+The `.yml` files in `.github/worflows` are workflow templates which are run on GitHub in response to specific actions.  The majority of workflows in this directory are Arrow implementation-specific and are run when changes are made which affect code relevant to that language's implementation, but other workflows worth noting are:
```

Comment:
I wouldn't use the template word here to avoid confusion with the actually jinja2 templated crossbow ymls.

Reply:
Good point!